internal/api/v20240610preview/hcpopenshiftclusters_methods.go

```diff
@@ -125,6 +126,18 @@ func (v version) NewHCPOpenShiftCluster(from *api.HCPOpenShiftCluster) api.Versi
 	return out
 }
 
+func newClusterCapabilitiesProfile(from *api.ClusterCapabilitiesProfile) *generated.ClusterCapabilitiesProfile {
+	out := &generated.ClusterCapabilitiesProfile{
+		Disabled: make([]*generated.OptionalClusterCapability, len(from.Disabled)),
+	}
+
+	for index, item := range from.Disabled {
+		out.Disabled[index] = api.Ptr(generated.OptionalClusterCapability(item))
+	}
+
+	return out
+}
```

Comment:
note to self: `api.StringSliceToStringPtrSlice` would not work here as is because of the string subtypes, but look into making it generic with `~string` constraints for the input and output types.